lib/broadway_kafka/producer.ex

```diff
@@ -575,6 +581,33 @@ defmodule BroadwayKafka.Producer do
     :ok
   end
 
+  @spec fetch_kafka_lag(pid()) ::
+          {:ok,
+           [
+             %{
+               topic: String.t(),
+               offsets: [
+                 {:ok,
+                  %{
+                    partition_index: non_neg_integer(),
+                    lag: non_neg_integer(),
+                    committed_offset: non_neg_integer(),
+                    partition_offset: non_neg_integer()
+                  }}
+                 | {:error, any()}
+               ]
+             }
+           ]}
+          | {:error, any()}
+  @doc """
+  Fetch the current offset of processed messages and the total number of messages
+  (`partition_offset`) for each partition and each topic the producer is consuming. Lag is the
+  difference between them. Calling this can be slow.
+  """
+  def fetch_kafka_lag(pid) do
```

Comment:
How would you get the PID that you pass here? Maybe we should include a tiny snippet that shows so?

Reply:
pushed